### DISCUSSION TITLE
PrismaClient - per process or per request?

### DISCUSSION THREAD

**Discussion by adarnon** [prisma/prisma#3776] [Q&A] [answered]
I am running a Next.js + GQL server, deployed on Heroku.
Through Prisma's documentation, I couldn't find a definite answer whether I should create one PrismaClient instance globally that would be shared for all requests, or create & destroy one for every request.
I tried creating an instance per request, bug I've encountered some problems. Looks like PrismaClient's `$disconnect` method destroys the engine process that is globally shared.
Shouldn't there be a way to manage connections on a per-request basis?

**Comment by ryands17** [comment, marked as answer]
Hey @adarnon 👋 

The best way to instantiate PrismaClient would be to create a single instance globally and use that single instance. As you are deploying on Heroku which is a long-running server, it would be best to have a single instance as that will manage your connections and you don't need to _connect/disconnect_ on every request.

We will add that to the docs as well.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
